pkg/retrieval/retrieval.go

```diff
@@ -87,16 +94,21 @@ func (s *Service) Protocol() p2p.ProtocolSpec {
 }
 
 const (
-	maxPeers             = 5
 	retrieveChunkTimeout = 10 * time.Second
 
 	retrieveRetryIntervalDuration = 5 * time.Second
 )
 
-func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address) (swarm.Chunk, error) {
+func (s *Service) RetrieveChunk(ctx context.Context, addr swarm.Address, origin bool) (swarm.Chunk, error) {
 	s.metrics.RequestCounter.Inc()
 
 	v, err, _ := s.singleflight.Do(addr.String(), func() (interface{}, error) {
```

Comment:
if a chunk is already retrieved with origin false, does that singleflight here not imply that we would never try with `maxPeers=8`  if a origin true call happens at the same time?

Reply:
Good point, the origin flag should be part of the singleflight key.